### ISSUE TITLE
current_trace_id is always nil

### ISSUE THREAD

**Issue by ayrat555** [spandex-project/spandex_ecto#22]
Hi, Zach and the team!

I found a weird issue using spandex_ecto. It seems if `spandex_ecto` is used separately without `spandex_phoenix`, the traces are not submitted because  current_trace_id in https://github.com/spandex-project/spandex_ecto/blob/master/lib/spandex_ecto/ecto_logger.ex#L21 is always nil 

One workaround is 

```elixir
    :ok =
      :telemetry.attach(
        "spandex-query-tracer",
        [:engine, :repo, :query],
        &submit_trace/4,
        nil
      )

....

  defp submit_trace(arg1, arg2, arg3, arg4) do
    _ = Tracer.start_trace("query")

    SpandexEcto.TelemetryAdapter.handle_event(arg1, arg2, arg3, arg4)

    _ = Tracer.finish_trace()
  end
```

**Comment by zachdaniel** [comment]
Yeah, the ecto tracer just doesn't expect to be the one starting traces. I'd accept a PR that starts a trace if one isn't found, perhaps based on configuration.

**Comment by GregMefford** [comment]
I think normally you wouldn't want to have a trace that is _only_ for Ecto calls - normally, you would want to have a trace for whatever the overall operation is, and just add a span for each Ecto call. Does that make sense? Do you have a use-case where it makes sense to have traces that are _only_ the Ecto calls?

**Comment by GregMefford** [comment]
Oh! I looked through your linked issues on your project and I think I understand the problem. You're starting/finishing the overall trace in the Cowboy process instead of the Phoenix/Plug process, so the trace information needs to be propagated from one process to the other somehow.

There currently isn't a good/supported way to do that, but I have an experimental "hack that works," if you want to copy/paste the same idea into your app. https://github.com/spandex-project/spandex_example/pull/5/files#diff-53dbbcb8ea2253c0eed6eadc69437bb1ccdcc943e2daac93f964e5f45d13a2f8R15
Basically what it does is put a record in ETS in the Cowboy process and then look it up from the Phoenix/Plug process, then clean it up in the Cowboy process when it ends.

I will go ahead and close this issue because I think the issue is with propagation between processes and not actually a problem with `spandex_ecto`, but please feel free to continue discussing this specific problem here if you'd like to.
